src/dlfcn.c

```diff
@@ -196,6 +201,44 @@ static void save_err_ptr_str( const void *ptr, DWORD dwMessageId )
     save_err_str( ptr_buf, dwMessageId );
 }
 
+static HMODULE MyGetModuleHandleFromAddress( void *addr )
+{
+    static BOOL (WINAPI *GetModuleHandleExAPtr)(DWORD, LPCSTR, HMODULE *) = NULL;
+    static BOOL failed = FALSE;
+    HMODULE kernel32;

```

Comment:
The variable ` HMODULE kernel32;` is only used in the block starting at line 213 with `if( !failed && GetModuleHandleExAPtr == NULL )` and could be moved to that location.